### PR TITLE
Make SignTool Accept Jar files

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #509 

**Changes:**

- Configuration.cs
  - Just added a condition to check "Jar" files; The certificate will be informed on the overriding itemgroup.
- FileName.cs
  - Added a method to check if the file is ".jar"
- SignToolTests.cs
  - Added new test case to make sure the tool accepts and recognize ".jar" files
- Sample.jar
  - Input for test case mentioned above

### CODE REVIEW

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs [hunk 1]
```diff
@@ -165,6 +165,21 @@ private SignInfo ExtractSignInfo(string fileFullPath)
                     return SignInfo.Empty;
                 }
             }
+            else if (FileName.IsJar(fileFullPath))
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.SignTool/src/Configuration.cs]: Could this be generalized to more file types which do not need to zip/rezip? e.g.

* .js
* .ps1
* .psd1
* .psm1
* .psc1
* .msi
* .dylib

Comment (anurse) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: Is it possible to have the tool simply accept any file and if it doesn't have special processing for it (i.e. repacking VSIX, NuPkg, etc.) then just pass it up to the ESRP service? Then this tool doesn't have to be explicitly updated for every individual signable file.

Comment (jcagme) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: I think this is a great idea @anurse. @JohnTortugo we can talk about this if you need.

Comment (jcagme) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: I see two checks here:
1.  Check if the file needs some sort of processing i.e. VSIX, .nuget
2.  Check if the file is sign-able or not

So, if the file does not fall in 1 it goes directly to check 2, if it does we unpack (process) it and pass the contents to two.

We can check if ESRP has a list of files which are sign-able so we create some sort of white list. @mmitche @MattGal do you know who in ESRP we can talk to?

Comment (natemcmaster) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: https://microsoft.sharepoint.com/teams/codesigninfo/Wiki/Signable Files.aspx

Comment (jcagme) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: This helps!

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: > The code that extracts containers (nupkg/vsix) attempts to code sign all of their contents.

It actually doesn't try to sign all kinds of files, take a look here: https://github.com/dotnet/arcade/blob/423fdc992191605f7326632057446e5811ba418d/src/Microsoft.DotNet.SignTool/src/Configuration.cs#L274

> Is it possible to have the tool simply accept any file and if it doesn't have special processing for it (i.e. repacking VSIX, NuPkg, etc.) then just pass it up to the ESRP service? 

Yeah, I'll create another issue/PR for that.

Comment (natemcmaster) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: Presumably this check should change too . We often bundle other signable files in vsix/nupkg containers.

https://github.com/dotnet/arcade/blob/423fdc992191605f7326632057446e5811ba418d/src/Microsoft.DotNet.SignTool/src/FileSignInfo.cs#L34-L35

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: > how to specify what to exclude from code signing. 

@natemcmaster I created issue #524 for us discuss about improvements to the exclusion of files during signing.

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: > Presumably this check should change too.

Sure, make sense.

File: tests/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 2]
```diff
@@ -241,5 +241,32 @@ public void NestedContainer()
 
             TestCaseEpilogue(itemsToSign, signingInformation, signingOverridingInformation, expectedSigningList);
         }
+
+        [Fact]
+        public void AcceptAJarFile()
+        {
+            // List of files to be considered for signing
+            var itemsToSign = new string[] {
```
Comment (tmat) [inline tests/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: Code style: `{` on new line.

**Comment by JohnTortugo** [comment]
I'm closing this PR in favor of a future one that accepts mapping between `file extension` == `friendly certificate name`, which will include support for `.jar` files.

**Pull request closed without merging**